### PR TITLE
expose memstore through memstore:// config url.

### PR DESCRIPTION
**Pull request by jmuk**
**Release note**:

```release-note
NONE
```

---
This change is [Reviewable](https://reviewable.io/reviews/istio/mixer/1165)

### CODE REVIEW

**Comment by jmuk** [comment]
rebased. PTAL.

**Comment by jmuk** [comment]
ping @mandarjog

**Comment by jmuk** [comment]
@mandarjog @guptasu please review this; I want this merged for my further tasks -- and this shouldn't affect the current production, they will not be used anyways.

File: pkg/config/store/fsstore2.go [hunk 1]
```diff
@@ -157,59 +151,43 @@ func (s *fsStore2) readFiles() map[Key]*resource {
 
 func (s *fsStore2) checkAndUpdate() {
 	newData := s.readFiles()
-	s.mu.Lock()
```
Comment (mandarjog) [inline pkg/config/store/fsstore2.go]: is checkAndUpdate single threaded now ?
Why was this lock required before and not now?

Comment (jmuk) [reply pkg/config/store/fsstore2.go]: The lock has been moved to the memstore, fsstore2 relies on it.

The lock here was for controls of writing in `checkAndUpdate` and reading in `List`, now both are controlled in memstore internally used by fsstore2; the new `Put` and `Delete` below handles the lock internally.

**Pull request merged**